### ISSUE TITLE
Office/Outlook 365 will no longer support Basic Auth by Oct 1st, 2022

### ISSUE THREAD

**Issue by Micraow** [deltachat/deltachat-android#2262]
- Android version: 10
- Device: redmi note 7
- Delta Chat version: 1.28.3
- Expected behavior: Receive mail normally.
- Actual behavior:Unable to receive mail normally. Login failed, but mail can be sent.
- Steps to reproduce the problem:Log in to office 365 or outlook mailbox using IMAP protocol.
- Screenshots:
- Logs:

I have no error log because I have enabled IMAP protocol manually. However, I learned from Microsoft that the IMAP protocol will be completely closed and cannot be re enabled by October 2022.This will affect the recipients of all outlook users who use Delta chat.At present, there is no appropriate solution. Their official suggestion is to use modern verification.

See here: https://techcommunity.microsoft.com/t5/exchange-team-blog/basic-authentication-and-exchange-online-september-2021-update/ba-p/2772210

I feel sad about it. I wonder if delta chat can support office365 in the future.

Add: smtp protocol won't be affected . It means that we can still send emails.

**Comment by r10s** [comment]
> I have no error log because I have enabled IMAP protocol manually. 

so, as of now, IMAP works with Outlook for you (after manually having it enabled in some settings)?

**Comment by Micraow** [comment]
Yes, it works normally now.

p.s.  I just looked at the relevant notice and found that the users of personal outlook may not be affected, but I am sure that the users of office 365 will be affected.

**Comment by gerryfrancis** [comment]
@Micraow Thank you for your report.

> I feel sad about it. I wonder if delta chat can support office365 in the future.

The article points out that "Basic Auth" can be reenabled by the user, and opt-outing from this procedure can be requested, too. Therefore I see no issue that Outlook 365 would not comply with clients using basic authentification, like Delta Chat does.

**Comment by Micraow** [comment]
> @Micraow Thank you for your report.
> 
> > I feel sad about it. I wonder if delta chat can support office365 in the future.
> 
> The article points out that "Basic Auth" can be reenabled by the user, and opt-outing from this procedure can be requested, too. Therefore I see no issue that Outlook 365 would not comply with clients using basic authentification, like Delta Chat does.

I know, but it can only be enabled temporarily.

There is a sentence in the notice:

starting October 1, 2022, we’ll begin turning off Basic Auth in all tenants, regardless of usage.

I think it means that on October 1, 2022, all users can no longer use IMAP.

(I come from a non English speaking country and English is not my mother tongue. If I have a wrong understanding of this notice, please point out that ，maybe I just understand the wrong meaning.)

**Comment by gerryfrancis** [comment]
The article says:

> In 2022, as we roll out the changes necessary to support this effort, we will begin disabling Basic Auth for some customers with usage on a short-term and temporary basis.

So I think Microsoft mean the performance of disabling Basic Auth itself to be temporary. And still you have the possibility to opt out from this action, so before they decide to apply it again, just opt out and you should not be bothered again.

**Comment by gerryfrancis** [comment]
@Micraow I believe there is nothing we could or should do, so I am closing this issue. In case you feel closing this issue was not justified, please let us know, thank you.

**Comment by Micraow** [comment]
Eh, but i think here's something that we should know:
At the end of this article, there is such a Q & A:

> How can I get a longer exception? I still want to use Basic Auth after October 2022

> We are not providing the ability to use Basic Auth after October 2022. You should ensure your dependency on Basic Auth in Exchange Online has been removed by that time.

**Comment by Micraow** [comment]
I hope I don't have to worry...

Thank you for your reply.

**Comment by gerryfrancis** [comment]
@Micraow You are right, I should have read the whole article, therefore reopening and renaming your issue adequately. :)

Microsoft 365 and Outlook.com support the OAuth 2.0 protocol, and so does Delta Chat. You could give it a try and report back here, we appreciate your valueable feedback in this regard. Thank you in advance!

**Comment by Micraow** [comment]
Can you tell me how to do it? I don't seem to have found the relevant options. I just changed the authentication mode from automatic to OAuth. However, it doesn't work.Prompt failed to get token.

Here is my log:
https://pastebin.com/rwtgraEz

**Comment by Gh0stz0x** [comment]
@Micraow 
Is it a free service or is it paid? 
Does it have IMAP?
If it does not block foreign IPs, I try to create an account and see if there is any need to enter particular settings 🤔

**Comment by Micraow** [comment]
In most cases, it's paid, but as a developer, you can easily get it for free.
See here: https://developer.microsoft.com/microsoft-365/dev-program

Yes, it has IMAP, and can work well.
It doesn't block any IPs. After you create a subscription , just create a user at admin.microsoft.com and test it.

Moreover, if you need a test mailbox, I am willing to offer one.

**Comment by Micraow** [comment]
> @Micraow Is it a free service or is it paid? Does it have IMAP? If it does not block foreign IPs, I try to create an account and see if there is any need to enter particular settings 🤔

I just misunderstood you.163 mail is free. And it has IMAP, but it needs some additional settings (like QQ mail). It doesn't block foreign IPs
